services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -214,6 +232,68 @@ class HakemusService(
         return hakemusRepository.save(hakemus).toHakemus()
     }
 
+    /**
+     * Create a new cable report application for an excavation announcement that has a cable report
+     * applied for.
+     *
+     * @return the new cable report application entity or null if no new application was applied for
+     */
+    private fun newJohtoselvitysAppliedFor(hakemus: HakemusEntity): HakemusEntity? {
+        val hakemusEntityData = hakemus.hakemusEntityData
+        val appliedFor =
+            when (hakemusEntityData) {
+                is KaivuilmoitusEntityData -> !hakemusEntityData.cableReportDone
+                else -> return null
+            }
+        if (!appliedFor) {
+            return null
+        }
```

Comment:
I think this would be cleaner, if the decision to create the accompanying johtoselvityshakemus and the creation itself would be in separate functions.

```suggestion
    private fun createAccompanyingJohtoselvityshakemus(hakemus: HakemusEntity): HakemusEntity? =
        when (val hakemusEntityData = hakemus.hakemusEntityData) {
            is KaivuilmoitusEntityData ->
                if (!hakemusEntityData.cableReportDone)
                    createAccompanyingJohtoselvityshakemus(hakemus, hakemusEntityData)
                else null
            else -> null
        }

    private fun createAccompanyingJohtoselvityshakemus(
        hakemus: HakemusEntity,
        hakemusData: KaivuilmoitusEntityData
    ): HakemusEntity {
```

Reply:
I splitted the code in separate functions as you suggested. You're right, it is more more cleaner that way.